### ISSUE TITLE
Consider gzipping the catalog index (Content-Encoding: gzip)

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/NuGetGallery#8013]
Currently the catalog index (https://api.nuget.org/v3/catalog0/index.json) is 1.91 MB. It will continue growing forever, giving the current design. It is unbounded in nature.

We can gzip it to make it faster to download. For example, gzipping locally gives the following compression level:
1.91 MB => 360 KB

This will make polling the catalog for changes faster.

**Comment by loic-sharma** [comment]
Good idea, but we should be careful to not break customers that are following the catalog today. Maybe we could do something similar to registration - we keep the old catalog index around forever, but, we only update the new catalog index going forward?

**Comment by joelverhagen** [comment]
Maybe. I was thinking just write the index twice. `catalog0/index.json` and `catalog1/index.json`. Catalog 1 could just point to Catalog 0 for non-gzipped blobs (i.e. everything but the index). We could extend the logic for pages too but it gets a bit more complicated.

We can edge cache the pages after they are full and leaves immediately so gzipping has smaller returns.
